apps/design/backend/src/file_storage_client.ts

```diff
@@ -0,0 +1,71 @@
+// eslint-disable-next-line max-classes-per-file
+import { Buffer } from 'node:buffer';
+import { createReadStream } from 'node:fs';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { Readable } from 'node:stream';
+import {
+  GetObjectCommand,
+  PutObjectCommand,
+  S3Client,
+} from '@aws-sdk/client-s3';
+import { assertDefined } from '@votingworks/basics';
+
+import { WORKSPACE } from './globals';
+
+export interface FileStorageClient {
+  readFile: (filePath: string) => Promise<Readable>;
+  writeFile: (filePath: string, contents: Buffer) => Promise<void>;
+}
+
+/**
+ * A file storage client backed by S3, for deployed environments
+ */
+export class S3FileStorageClient {
+  private readonly s3Client: S3Client;
+
+  constructor() {
+    this.s3Client = new S3Client({ region: process.env.AWS_S3_REGION });
+  }
+
+  async readFile(filePath: string): Promise<Readable> {
+    const data = await this.s3Client.send(
+      new GetObjectCommand({
+        Bucket: process.env.AWS_S3_BUCKET_NAME,
+        Key: filePath,
+      })
+    );
+    return (data.Body ?? Readable.from([])) as Readable;
```

Comment:
We might want to consider using `Result<Readable, FileStorageClientError>` in the future. I'm assuming `data.Body` is undefined or null when the file does not exist, so it's not quite dire enough to throw. But it might make debugging and error handling more confusing to silently return no data like this.

Reply:
I actually don't know if `data.Body` will be undefined when the file doesn't exist and am running out of time to test, but your suggestion re error handling is a good one! Set up the scaffolding here: https://github.com/votingworks/vxsuite/pull/5937/commits/53a62d70936e7952822d4605c02a85f87a25029d